modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/manager/CookiesManagerImpl.java

```diff
@@ -108,6 +108,22 @@ public boolean addCookie(
 			cookiesMap.put(StringUtil.toUpperCase(cookie.getName()), cookie);
 		}
 
+		if ((_knownCookies.get(cookie.getName()) != null) &&
+			(_knownCookies.get(cookie.getName()) != consentType)) {
+
+			if (_log.isWarnEnabled()) {
+				_log.warn(
+					"This cookie has been previously added with different consent type " +
+						cookie.getName());
+				_log.warn("Known consent type: " + consentType);
+				_log.warn(
+					"Current consent type: " +
+						_knownCookies.get(cookie.getName()));
+			}
+		} else {
+			_knownCookies.put(cookie.getName(), consentType);
+		}
+
```

Comment:
The new `_knownCookies` Map is system scoped. Entries never get removed.  Which might be fine because it is unlikely to grow large. 

But what if modules get upgraded, changing the category of a cookie? 
I think we can avoid a need to restart the module if we change the logic to update _knownCookies with an new category after logging that message. What do you think?

p.s. I can see the current logic being fine for an initial release also because the concern only becomes release when modules need to evolve over time. So we could potentially handle this in a separate bug ticket.

Reply:
Great suggestion! Thank you! I'll change it.